File: server/src/main/java/moe/yuuta/server/api/ApiUtils.java
package moe.yuuta.server.api;

import com.fasterxml.jackson.core.JsonProcessingException;
import com.fasterxml.jackson.core.type.TypeReference;
import com.fasterxml.jackson.databind.ObjectMapper;

import java.io.IOException;
import java.util.List;

public class ApiUtils {
    public static String objectToJson (Object object) throws JsonProcessingException {
        return new ObjectMapper().writeValueAsString(object);
    }

    public static String tryObjectToJson (Object object) {
        try {
            return objectToJson(object);
        } catch (JsonProcessingException e) {
            return null;
        }
    }

    public static <V> V jsonToObject (String json, Class<V> t) throws IOException {
        return new ObjectMapper().readValue(json, t);
    }

    public static <V> V jsonToObject (String json, TypeReference<V> t) throws IOException {
        return new ObjectMapper().readValue(json, t);
    }

    public static String separateListToComma(List<String> list) {
        StringBuilder builder = new StringBuilder();
        for (String value : list) {
            builder.append(value);
            builder.append(",");
        }
        String values = builder.toString();
        values = values.substring(0, values.length() - 1);
        return values;
    }
}

File: server/src/main/kotlin/moe/yuuta/server/api/ApiUtils.kt
package moe.yuuta.server.api

import com.fasterxml.jackson.core.JsonProcessingException
import com.fasterxml.jackson.core.type.TypeReference
import com.fasterxml.jackson.databind.ObjectMapper

import java.io.IOException

object ApiUtils {
    @JvmStatic
    @Throws(JsonProcessingException::class)
    fun objectToJson(obj: Any?): String {
        return ObjectMapper().writeValueAsString(obj)
    }

    @JvmStatic
    fun tryObjectToJson(obj: Any?): String? {
        try {
            return objectToJson(obj)
        } catch (e: JsonProcessingException) {
            return null
        }
    }

    @JvmStatic
    @Throws(IOException::class)
    fun <V> jsonToObject(json: String, t: Class<V>): V {
        return ObjectMapper().readValue(json, t)
    }

    @JvmStatic
    @Throws(IOException::class)
    fun <V> jsonToObject (json: String, t: TypeReference<V>): V {
        return ObjectMapper().readValue(json, t)
    }

    @JvmStatic
    fun separateListToComma(list: List<String>): String {
        val builder = StringBuilder()
        for (value in list) {
            builder.append(value)
            builder.append(",")
        }
        var values = builder.toString()
        values = values.substring(0, values.length - 1)
        return values
    }
}
